Automattic/jetpack
# Plugin Installer: remove excessive dependencies

## Proposed changes:
* Extract the `is_current_request_activating_plugin_from_plugins_screen()` method from Plugin Installer to Status package.
* Remove explicit Plugin Installer package dependencies from standalone plugins (it's still required by My Jetpack though).
* Add explicit Status package dependency to standalone plugins (it's already loaded by other packages though).

### Other information:


## Jetpack product discussion
https://github.com/Automattic/vulcan/issues/334

## Does this pull request change what data or activity we track or use?
No.

## Testing instructions:
1. Go to "Plugins" page and activate one of the affected plugins (e.g. Jetpack VaultPress Backup).
2. Confirm you get redirected to the plugin's settings page after plugin activation.
3. Go to "My Jetpack" and activate another affected plugin (e.g. Jetpack Protect) from there. Confirm it got activated without page reload.
